sdk/core/System.ClientModel/tests/Samples/ModelReaderWriterSamples.cs

```diff
@@ -58,6 +58,90 @@ public void Stj_Read_Simple()
         #endregion
     }
 
+    public void Write_Proxy()
+    {
+        #region Snippet:Readme_Write_Proxy
+        InputModel model = new InputModel();
+
+        ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+        options.AddProxy(new InputModelProxy());
+
+        BinaryData data = ModelReaderWriter.Write(model, options);
+        #endregion
+    }
+
+    public void Read_Proxy()
+    {
+        #region Snippet:Readme_Read_Proxy
+        string json = @"{
+              ""x"": 1,
+              ""y"": 2,
+              ""z"": 3
+            }";
+
+        ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+        options.AddProxy(new OutputModelProxy());
+
+        OutputModel? model = ModelReaderWriter.Read<OutputModel>(BinaryData.FromString(json), options);
+        #endregion
+    }
+
+    private class OutputModelProxy : IJsonModel<OutputModel>
```

Comment:
Consider adding this class declaration somewhere in the sample -- it was very helpful for me to see this to understand how proxies actually work.